### DISCUSSION TITLE
KVM support?

### DISCUSSION THREAD

**Discussion by brainer3220** [termux/termux-app#2594] [Q&A] [answered]
[Recently report](https://twitter.com/kdrag0n/status/1492712401262710784?s=20&t=E3d9YLa7wdRa3IX3Igttug), Android 13 is available KVM, and can support KVM based termux?

**Comment by ghost** [comment, marked as answer]
QEMU has [kvm enabled](https://github.com/termux/termux-packages/blob/192876115260909009026733f29af3a2df8d76f9/packages/qemu-system-x86-64-headless/build.sh#L87) though you need to give Termux a KVM access by running `qemu` as `root` or global writable access to `kvm` device file `chmod 666 /dev/kvm` (needs root)

***

Great concept but not all devices can update to Android 13 or vendors may not allow it at all so we just wait until Google officially announces about virtualization for Android 13, things may change. `proot` still be a solution if running a rootless Linux distributions on Android
